nevergrad/functions/causaldiscovery/test_core.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import nevergrad as ng
+from . import core
+
+
+def test_causal_discovery_using_data() -> None:
+    func = core.CausalDiscovery(generator='sachs')
+    assert func._nvars == 11
+
+
+def test_causal_discovery_using_generator() -> None:
+    nnodes = 13
+    npoints = 55
+    func = core.CausalDiscovery(generator='acylicgraph', npoints=npoints, nodes=nnodes)
+    assert func._nvars == nnodes
+    assert func._data.shape == (npoints, nnodes)
+    assert func.graph_score(func._ground_truth_graph) == 1
+    optimizer = ng.optimizers.NGO(parametrization=func.parametrization, budget=500)
+    recommendation = optimizer.minimize(func.function)
+    assert func.objective(**recommendation.kwargs) == func._min_score_so_far #Why sometimes False?
```

Comment:
good question, i this is flaky we need to know why.
The answer may come from the parametrization, if it includes Choice variables, then it's noisy, so NGO may choose a noisy optimizer, which may provide an average for best recommendation. You should probably use OnePlusOne instead to avoid any issue.

Reply:
NGOpt has changed quite a bit recently about these Choice variables, so maybe today the situation is quite different.